### PR TITLE
[FLINK-23052][ci] Improve stability of maven snapshot deployment

### PR DESCRIPTION
**Pull request by rmetzger**
The deployment was timing out or failing a lot recently.

### CODE REVIEW

File: tools/azure-pipelines/build-nightly-dist.yml [hunk 1]
```diff
@@ -109,7 +109,7 @@ jobs:
             </settings>
             EOF
 
-            export CUSTOM_OPTIONS="-Dgpg.skip -Drat.skip -Dcheckstyle.skip --settings $(pwd)/deploy-settings.xml"
+            export CUSTOM_OPTIONS="-Dgpg.skip -Drat.skip -Dcheckstyle.skip -Dmaven.wagon.http.pool=false --settings $(pwd)/deploy-settings.xml"
```
Comment (nicoweidner) [inline tools/azure-pipelines/build-nightly-dist.yml:112]: What is the reason for this change? Are we creating more connections than the pool can handle by default?

Comment (rmetzger) [reply tools/azure-pipelines/build-nightly-dist.yml:112]: Nobody knows: https://stackoverflow.com/a/58769870/568695 .. it's the black magic needed to make a very old maven version work on CI ;-)

Comment (nicoweidner) [reply tools/azure-pipelines/build-nightly-dist.yml:112]: Alright, if it works... :-D

**Review by nicoweidner — APPROVED** [review]

File: tools/azure-pipelines/build-nightly-dist.yml (hunk 1 above)
Comment (rmetzger) [reply tools/azure-pipelines/build-nightly-dist.yml:112]: I can not guarantee that it works, but at least in my dry run, it did: https://dev.azure.com/rmetzger/Flink/_build/results?buildId=9150&view=logs&j=eca6b3a6-1600-56cc-916a-c549b3cde3ff

**Comment by rmetzger** [comment]
Thanks for your review!
Merging ...
